### PR TITLE
add xavier's leaky RRXOR

### PR DESCRIPTION
**Pull request by casperlchristensen**
Taking @14xp 's implementation of leaky RRXOR and creating a new branch (since the old branch is a bit stale with all the new changes).

---

> [!NOTE]
> Introduces `leaky_rrxor` transition matrices, registers it in HMM functions, and adds unit tests validating behavior and epsilon edge case.
> 
> - **Generative Processes**:
>   - Add `leaky_rrxor(p1, p2, epsilon)` implementing a leaky variant of `rrxor` by mixing with a uniform leak.
>   - Register `"leaky_rrxor"` in `HMM_MATRIX_FUNCTIONS`.
> - **Tests**:
>   - Add `test_leaky_rrxor` to validate shape, HMM properties, and deviation from base `rrxor` per `epsilon`.
>   - Add `test_leaky_rrxor_zero_epsilon` to assert equality with `rrxor` when `epsilon=0`.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 0c9a6606e44a122808970e6b6eb2a04b3dd0215d. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: tests/generative_processes/test_transition_matrices.py [hunk 1]
```diff
@@ -171,6 +172,27 @@ def test_rrxor():
     assert jnp.allclose(stationary_distribution, jnp.array([2, 1, 1, 1, 1]) / 6)
 
 
+def test_leaky_rrxor():
```
[the remark this answers is not in the document]
Comment (ealt) [reply tests/generative_processes/test_transition_matrices.py]: @claude The tests should be ordered alphabetically by process, "leaky_rrxor" should come between "fanizza" and "matching_parens"

**Review by ealt — APPROVED** [review]
